packages/bot-web-ui/src/pages/bot-builder/bot-builder.tsx

```diff
@@ -85,16 +84,12 @@ const BotBuilder = observer(() => {
     }, [is_loading]);
 
     const handleBlockDelete = (e: TBlocklyEvents) => {
-        if (e.isUiEvent && e.type === 'selected' && !e.group?.includes('dbot-')) {
-            selection_id = e.newElementId;
-        }
-
-        if (e.type === 'drag' && !e.isStart && !e.group?.includes('dbot-')) {
-            end_drag_id = e.group;
+        const { is_reset_button_clicked } = toolbar;
+        if (e.type === 'delete' && !is_reset_button_clicked) {
+            deleted_block_id = e.blockId;
         }
-        if (e.type === 'delete' && (end_drag_id === e.group || selection_id === e.blockId)) {
+        if (e.type === 'selected' && deleted_block_id === e.oldElementId) {
```

Comment:
@farabi-deriv is this `oldElementId` coming from the event?

Reply:
Yes @shafin-deriv